### PR TITLE
Add an exact match check to equiv TitleSubset scorer.

### PR DESCRIPTION
**Pull request by TjlHope**
If there is a name consisting of purely common words, or stripped
characters then titles won't match. In this case if there's an exact
match it would make sense to return a positive score.
This scorer also originally replaced an exact match scorer, so it
simply makes it a complete replacement/improvement of the old scorer.

### CODE REVIEW

File: src/test/java/org/atlasapi/equiv/scorers/TitleSubsetBroadcastItemScorerTest.java [hunk 1]
```diff
@@ -45,6 +45,10 @@ public void testMatches() {
                 itemWithTitle("Power - Rangers: R.P.M.!!"),
                 itemWithTitle("Power Rangers RPM")
         ));
+        assertEquals(Score.ONE, score(
```
Comment (AdamDM) [inline src/test/java/org/atlasapi/equiv/scorers/TitleSubsetBroadcastItemScorerTest.java]: As this tests for a unique thing it should be it's own test (as should the others originally)

File: src/main/java/org/atlasapi/equiv/scorers/TitleSubsetBroadcastItemScorer.java [hunk 2]
```diff
@@ -83,52 +82,62 @@ protected String getName() {
...
+    private boolean titleMissing(Content subject) {
+        return Strings.isNullOrEmpty(subject.getTitle());
+    }
+
+    private String sanitize(String title) {
+        return titleTransformer.expand(title).toLowerCase();
+    }
+
+    private boolean titlesEqual(String subjectTitle, String candidateTitle) {
+        return false && subjectTitle.equals(candidateTitle);
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/equiv/scorers/TitleSubsetBroadcastItemScorer.java]: wouldn't false and something always be false?

Comment (TjlHope) [inline src/main/java/org/atlasapi/equiv/scorers/TitleSubsetBroadcastItemScorer.java]: Oh bollocks - added that in to test the test something else - completely forgot to remove it - test won't actually pass at the moment - doh!

File: src/test/java/org/atlasapi/equiv/scorers/TitleSubsetBroadcastItemScorerTest.java (hunk 1 above)
Comment (TjlHope) [inline src/test/java/org/atlasapi/equiv/scorers/TitleSubsetBroadcastItemScorerTest.java]: Fair point, I'll split them up

**Pull request merged**